docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,237 @@
+
+- for each page configuration, the configured component is added as a dependency to the template app module
+- if the page configuration contains a templateName, a dependency to that template view is added to the app
+- otherwise, the class definition of the component is analyzed to find a default template view name. if found, a dependency to that view is added to the app module
+
+The template component is analyzed in the following way:
+
+- precondition: template component class is defined in an AMD-style module, using define or sap.ui.define
+- precondition: the module 'sap/suite/ui/generic/template/lib/TemplateAssembler' is imported
+- precondition: a call to TemplateAssembler.getTemplateComponent is used to define the component class
+- precondition: that call is used in a top level return statement of the factory function
+- precondition: necessary parameters to that call are given as an object literal (no further coding)
+- precondition: the settings define a managed property property 'metadata.properties.templateName' with a defaultValue of type string
+The default value of the property represents the template view of the template component.
+The manifest of the template app in theory could specify an alternative template in
+component.settings.templateName.
+
+### XML Template Analyzer
+
+The [XML Template Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/XMLTemplateAnalyzer.js) tackles XMLViews and XMLFragments. It parses the XML, collects controls and adds them as dependency to the ModuleInfo object.
+Additionally, some special dependencies are handled:
+
+- controller of the view
+- resource bundle (note: `sap/ui/core/Locale.js` dependent dependencies can't be modeled yet in ModuleInfo)
+- component referenced via ComponentContainer control
+- embedded fragments or views
+
+In an XMLView/XMLFragment, there usually exist 3 categories of element nodes: controls, aggregations
+of cardinality 'multiple' and non-UI5 nodes (e.g. XHTML or SVG). The third category usually
+can be identified by its namespace. To distinguish between the first and the second
+category, this analyzer uses a ResourcePool (provided by the caller and usually derived from the
+library classpath). When the qualified node name is contained in the pool, it is assumed to
+represent a control, otherwise it is ignored.
+
+In an XMLView/XMLFragment, aggregations of cardinality 0..1 (single) also appear as element nodes in the XML, they just must not contain more than one element.
+
+In certain cases this might give wrong results, but loading the metadata for each control
```

Comment:
"In certain cases this ..."

The context of "this" is not clear as there's now a paragraph between the "this" (usage of pool to determine whether an XML node is a control or an aggregation) and the sentence here.

I would either move the aggregation part elsewhere (maybe I would even completely omit the talk about cardinalities) or a short form of "this" has to be repeated again here, .e.g. "this check against the set of known controls..."

Reply:
completely removed the content when talking about cardinalities